src/Controller/PayplugGenericGateway.php

```diff
@@ -163,6 +163,7 @@ public function check_billing_country_permissions($account, $billing_code){
 	private function process_standard_intent_payment($order){
 
 		if ( !is_wc_endpoint_url('order-pay') &&
+			empty($_POST["payplug_non_blocks"]) &&
 			$this->is_checkout_block() &&
```

Comment:
😅 I dont get this if hehe what I understand from here is that $this->is_checkout_block() is not working well, so you need to send the param on the front, but on the other hand if the is_checkout_block is not working should be removed and left only $_POST["payplug_non_blocks"] instead

Reply:
or maybe I'm missing something

Reply:
that's exacly it! 

`$this->is_checkout_block()` - true or false if checkout page has blocks enabled.

The problem is: on order-review-page blocks is not enabled. so it will enter this `if` even though we don't want it.

The easiest way for us to assess if it should enter or not is: is he using the script we done for non-block order?  if he is not, that he should enter here, if he's he shouldn't, so whenever we use the script we always send this.